eZ/Publish/Core/Repository/ContentTypeService.php

```diff
@@ -934,11 +934,14 @@ public function loadContentTypeList(array $contentTypeIds, array $prioritizedLan
     /**
      * {@inheritdoc}
      */
-    public function loadContentTypes(APIContentTypeGroup $contentTypeGroup, array $prioritizedLanguages = [])
-    {
+    public function loadContentTypes(
+        APIContentTypeGroup $contentTypeGroup,
+        array $prioritizedLanguages = [],
+        int $status = SPIContentType::STATUS_DEFINED
```

Comment:
Something doesn't add up here. On the interface you have:
```php
public function loadContentTypes(
    ContentTypeGroup $contentTypeGroup,
    array $prioritizedLanguages = [],
    int $status = ContentType::STATUS_DEFINED
): array
```
where `ContentType` is of `\eZ\Publish\API\Repository\Values\ContentType\ContentType`
while here you're using SPI Value Object.
I see handling of that is inconsistent in several places in this class. API implementation should accept constant from API Value Object. SPI/Persistence should accept constants from SPI/Persistence value object. Therefore IMHO the safe course of action would be to define some sort of mapping. Even though right now it's gonna be 1:1, It's not guaranteed to be always like that.

Reply:
@alongosz I've changed the `$status` argument to API's one, however, what do you mean by defining additional mapping - is this really required here?

Reply:
> what do you mean by defining additional mapping - is this really required here?

There's no guarantee that API status will be always aligned with SPI/Persistence status. But ATM no real value and query-wise safe-ish because of strict types.